api/jwt/jwt.go

```diff
@@ -82,18 +91,30 @@ func (service *Service) ParseAndVerifyToken(token string) (*portainer.TokenData,
 	if err == nil && parsedToken != nil {
 		if cl, ok := parsedToken.Claims.(*claims); ok && parsedToken.Valid {
 			tokenData := &portainer.TokenData{
-				ID:       portainer.UserID(cl.UserID),
-				Username: cl.Username,
-				Role:     portainer.UserRole(cl.Role),
+				ID:         portainer.UserID(cl.UserID),
+				Username:   cl.Username,
+				Role:       portainer.UserRole(cl.Role),
+				OAuthToken: cl.OAuthToken,
 			}
+
+			if tokenData.OAuthToken != "" {
+				if service.tokenBlacklist.IsBlocked(tokenData.OAuthToken) {
+					return nil, errInvalidJWTToken
+				}
+			}
+
 			return tokenData, nil
 		}
 	}
-
 	return nil, errInvalidJWTToken
 }
 
 // SetUserSessionDuration sets the user session duration
 func (service *Service) SetUserSessionDuration(userSessionDuration time.Duration) {
 	service.userSessionTimeout = userSessionDuration
 }
+
+// AddTokenToBlacklist adds a token identifier to the token list
+func (service *Service) AddTokenToBlacklist(token string) {
```

Comment:
Woring as above:

I suggest to use `AddTokenToBlocklist` in order to use a welcoming and inclusive language.

Reply:
Great finding, thanks :+1: